src/rendering/webgl/programs/common/program.ts

```diff
@@ -121,59 +128,128 @@ export abstract class Program<Uniform extends string = string> implements Abstra
       if (location === -1) throw new Error(`Program: error while getting location for attribute "${attr.name}".`);
       this.attributeLocations[attr.name] = location;
     });
+
+    // For instanced programs:
+    if ("CONSTANT_ATTRIBUTES" in definition) {
+      this.isInstanced = true;
+
+      const constantAttributesItemsCount = getAttributesItemsCount(definition.CONSTANT_ATTRIBUTES);
+
+      if (definition.CONSTANT_DATA.length !== this.VERTICES)
+        throw new Error(
+          `Program: error while getting constant data (expected ${this.VERTICES} items, received ${definition.CONSTANT_DATA.length} instead)`,
+        );
+
+      const constantData: number[] = [];
+      for (let i = 0; i < definition.CONSTANT_DATA.length; i++) {
+        const vector = definition.CONSTANT_DATA[i];
+
+        if (vector.length !== constantAttributesItemsCount)
+          throw new Error(
+            `Program: error while getting constant data (one vector has ${vector.length} items instead of ${constantAttributesItemsCount})`,
+          );
+
+        constantData.push(...vector);
```

Comment:
I guess the length of this vector cannot overflow JS stack size regarding maximum function arity (I think so but just making sure)?